hibernate-core/src/main/java/org/hibernate/procedure/internal/StandardCallableStatementSupport.java

```diff
@@ -47,86 +45,41 @@ public StandardCallableStatementSupport(boolean supportsRefCursors) {
 	public JdbcCall interpretCall(
 			String procedureName,
 			FunctionReturnImpl functionReturn,
-			ParameterMetadataImplementor parameterMetadata,
+			ProcedureParameterMetadataImplementor parameterMetadata,
 			ProcedureParamBindings paramBindings,
 			SharedSessionContractImplementor session) {
-		final StringBuilder buffer = new StringBuilder().append( "{call " )
+		final List<? extends ProcedureParameterImplementor<?>> registrations = parameterMetadata.getRegistrationsAsList();
+		final StringBuilder buffer = new StringBuilder(9 + procedureName.length() + registrations.size() * 2).append( "{call " )
 				.append( procedureName )
 				.append( "(" );
 
-		parameterMetadata.visitParameters(
-				new Consumer<QueryParameterImplementor<?>>() {
-					String sep = "";
-
-					@Override
-					public void accept(QueryParameterImplementor<?> param) {
-						if ( param == null ) {
-							throw new QueryException( "Parameter registrations had gaps" );
-						}
+		String sep = "";
+		for ( int i = 0; i < registrations.size(); i++ ) {
+			if ( registrations.get( i ).getMode() == ParameterMode.REF_CURSOR ) {
+				verifyRefCursorSupport( session.getJdbcServices().getJdbcEnvironment().getDialect() );
+				buffer.append( sep ).append( "?" );
+				sep = ",";
+			}
+			else {
+				buffer.append( sep ).append( "?" );
+				sep = ",";
+			}
+		}
 
-						final ProcedureParameterImplementor parameter = (ProcedureParameterImplementor) param;
+		buffer.append( ")}" );
 
-						if ( parameter.getMode() == ParameterMode.REF_CURSOR ) {
-							verifyRefCursorSupport( session.getJdbcServices().getJdbcEnvironment().getDialect() );
-							buffer.append( sep ).append( "?" );
-							sep = ",";
-						}
-						else {
-							throw new NotYetImplementedFor6Exception( getClass() );
-//							parameter.getHibernateType().visitJdbcTypes(
-//									sqlExpressableType -> {
-//										buffer.append( sep ).append( "?" );
-//										sep = ",";
-//									},
-//									Clause.IRRELEVANT,
-//									session.getFactory().getTypeConfiguration()
-//							);
-						}
-					}
-				}
-		);
+		return new JdbcCallImpl.Builder(
+				buffer.toString(),
+				parameterMetadata.hasNamedParameters() ?
+						ParameterStrategy.NAMED :
+						ParameterStrategy.POSITIONAL
+		).buildJdbcCall();
 
-		throw new NotYetImplementedFor6Exception( getClass() );
-//		return buffer.append( ")}" ).toString();
 	}
 
 	private void verifyRefCursorSupport(Dialect dialect) {
 		if ( ! supportsRefCursors ) {
 			throw new QueryException( "Dialect [" + dialect.getClass().getName() + "] not known to support REF_CURSOR parameters" );
 		}
 	}
-
-	@Override
-	public void registerParameters(
-			String procedureName,
-			CallableStatement statement,
-			ParameterStrategy parameterStrategy,
-			ParameterMetadataImplementor parameterMetadata,
-			SharedSessionContractImplementor session) {
-		throw new NotYetImplementedFor6Exception( getClass() );
-
-//		final AtomicInteger count = new AtomicInteger( 1 );
```

Comment:
Another commented-out block.

Plus, throws NotYetImplemented exception.  Is this still pending?

Reply:
Ignore my "block comments" comment.  Was just being blind.  Still early ;)